source/MaterialXView/ShaderValidators/Osl/OslValidator.cpp

```diff
@@ -43,23 +44,111 @@ void OslValidator::initialize()
+    // "%shader_output%" string siwht shader name and shader output
+    // respectively. Write to local file to use as input for rendering.
+    //
+    std::ifstream sceneTemplateStream(_oslTestRenderSceneTemplateFile);
+    std::string sceneTemplateString;
+    sceneTemplateString.assign(std::istreambuf_iterator<char>(sceneTemplateStream),
+        std::istreambuf_iterator<char>());
+
+    StringMap replacementMap;
+    replacementMap["%shader%"] = shaderName;
+    replacementMap["%shader_output%"] = outputName;
+    std::string sceneString = replaceSubstrings(sceneTemplateString, replacementMap);
+    if ((sceneString == sceneTemplateString) || sceneTemplateString.empty())
+    {
+        const std::string errorType("OSL rendering error.");
+        ShaderValidationErrorList errors;
+        errors.push_back("Scene template file: " + _oslTestRenderSceneTemplateFile + 
+                         "does not include proper tokens for rendering");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+
+    // Write scene file
+    const std::string sceneFileName(shaderPath + "_scene.xml");
+    std::ofstream shaderFileStream;
+    shaderFileStream.open(sceneFileName);
+    if (shaderFileStream.is_open())
+    {
+        shaderFileStream << sceneString;
+        shaderFileStream.close();
+    }
+
+    // Set oso file paths
+    std::string osoPaths(_oslUtilityOSOPath);
+    osoPaths += ";" + outputPath;
+
+    // Build and run render command
+    //
+    std::string command(_oslTestRenderExecutable);
+    command += " " + sceneFileName;
+    command += " " + outputFileName;
+    command += " --path " + osoPaths;
+    command += " > " + errorFile + redirectString;
+
+    int returnValue = std::system(command.c_str());
+
+    std::ifstream errorStream(errorFile);
+    std::string result;
+    result.assign(std::istreambuf_iterator<char>(errorStream),
+        std::istreambuf_iterator<char>());
+
+    if (!result.empty())
+    {
+        const std::string errorType("OSL rendering error.");
+        ShaderValidationErrorList errors;
+        errors.push_back("Command string: " + command);
+        errors.push_back("Command return code: " + std::to_string(returnValue));
+        errors.push_back("Shader failed to render:");
+        errors.push_back(result);
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+}
+
+void OslValidator::shadeOSL(const std::string& outputPath, const std::string& shaderName, const std::string& outputName)
 {
     // If no command and include path specified then skip checking.
     if (_oslTestShadeExecutable.empty() || _oslIncludePathString.empty())
     {
         return;
     }
 
+    FilePath shaderFilePath(outputPath);
+    shaderFilePath = shaderFilePath / shaderName;
+    std::string shaderPath = shaderFilePath.asString();
+
     // Set output image name. 
-    std::string outputFileName = shaderName + ".png";
+    std::string outputFileName = shaderPath + ".testshade.png";
```

Comment:
Minor changes since I keep output path and shader name in separate variables. Need to join them together now.